test/typeahead/typeahead_spec.js

```diff
@@ -476,16 +476,33 @@ describe('Typeahead', function() {
         expect(payload.preventDefault).not.toHaveBeenCalled();
       });
 
-      it('should autocomplete to top suggestion', function() {
-        var $el;
-
-        $el = $('<foo>');
-        spyOn(this.view, 'autocomplete');
-        this.menu.getTopSelectable.andReturn($el);
-
-        this.input.trigger(eventName, payload);
-
-        expect(this.view.autocomplete).toHaveBeenCalledWith($el);
+      it('should autocomplete to top suggestion on default (option tabAutocomplete not set)', function() {
+         var $el;
+         $el = $('<foo>');
+         spyOn(this.view, 'autocomplete');
+         this.menu.getTopSelectable.andReturn($el);
+         this.input.trigger(eventName, payload);
+         expect(this.view.autocomplete).toHaveBeenCalledWith($el);
```

Comment:
This test would be more durable if we could read the resulting HTML, as I'm doing with tests using enzyme... just because the view was called with the $el does not mean it actually autocompleted right?

Particularly with a loosely-maintained library like this, it's important to have reliable tests - I recommend writing an integration test https://github.com/corejavascript/typeahead.js/tree/master/test/integration

But we weren't able to revive integration tests...